diff -rupN gcc-4.8.0/config.sub gcc-4.8.0-patched/config.sub
--- gcc-4.8.0/config.sub	2013-05-27 00:03:04.720744716 +0200
+++ gcc-4.8.0-patched/config.sub	2013-05-27 00:03:42.190733185 +0200
@@ -1353,7 +1353,7 @@ case $os in
 	# The portable systems comes first.
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
-	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* | -pirix* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
diff -rupN gcc-4.8.0/gcc/config.gcc gcc-4.8.0-patched/gcc/config.gcc
--- gcc-4.8.0/gcc/config.gcc	2013-05-27 00:02:39.077419333 +0200
+++ gcc-4.8.0-patched/gcc/config.gcc	2013-05-27 00:03:42.194066517 +0200
@@ -805,6 +805,15 @@ case ${target} in
     *) echo 'Unknown thread configuration for VxWorks'; exit 1 ;;
   esac
   ;;
+*-*-pirix*)
+  extra_options="$extra_options gnu-user.opt"
+  extra_parts="crtbegin.o crtbeginS.o crtbeginT.o crtend.o crtendS.o libgcc_eh.a"
+  tm_defines="$tm_defines DEFAULT_LIBC=LIBC_GLIBC"
+  use_gcc_stdint=wrap
+  gas=yes
+  gnu_ld=yes
+  default_use_cxa_atexit=yes
+  ;;
 *-*-elf)
   # Assume that newlib is being used and so __cxa_atexit is provided.
   default_use_cxa_atexit=yes
@@ -933,6 +939,17 @@ arm*-*-eabi* | arm*-*-symbianelf* | arm*
 	esac
 	tm_file="${tm_file} arm/aout.h vxworks-dummy.h arm/arm.h"
 	;;
+arm-*-pirix*)
+	tm_file="dbxelf.h elfos.h gnu-user.h arm/unknown-elf.h arm/elf.h arm/aout.h arm/arm.h pirix.h arm/pirix.h glibc-stdint.h"
+	tmake_file="arm/t-arm arm/t-arm-elf"
+	tmake_file="${tmake_file} arm/t-arm-softfp soft-fp/t-softfp"
+	use_fixproto=yes
+	;;
+i[3-7]86-*-pirix*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h gnu-user.h i386/gnu-user-common.h i386/gnu-user.h pirix.h i386/pirix.h glibc-stdint.h"
+	tmake_file="${tmake_file} i386/t-i386elf t-svr4 i386/t-crtstuff"
+	use_fixproto=yes
+	;;
 avr-*-rtems*)
 	tm_file="elfos.h avr/elf.h avr/avr-arch.h avr/avr.h dbxelf.h avr/rtems.h rtems.h newlib-stdint.h"
 	tmake_file="avr/t-avr avr/t-multilib t-rtems avr/t-rtems"
diff -rupN gcc-4.8.0/libgcc/config.host gcc-4.8.0-patched/libgcc/config.host
--- gcc-4.8.0/libgcc/config.host	2013-05-27 00:03:27.937404226 +0200
+++ gcc-4.8.0-patched/libgcc/config.host	2013-05-27 00:03:42.200733181 +0200
@@ -354,6 +354,14 @@ arm*-*-eabi* | arm*-*-symbianelf* | arm*
 	tmake_file="$tmake_file t-softfp-sfdf t-softfp-excl arm/t-softfp t-softfp"
 	unwind_header=config/arm/unwind-arm.h
 	;;
+arm-*-pirix*)
+    extra_parts="crtbegin.o crtbeginS.o crtbeginT.o crtend.o crtendS.o libgcc_eh.a"
+    tmake_file="$tmake_file arm/t-arm arm/t-softfp t-softfp"
+    ;;
+i[3-7]86-*-pirix*)
+    extra_parts="crtbegin.o crtbeginS.o crtbeginT.o crtend.o crtendS.o libgcc_eh.a"
+    tmake_file="$tmake_file i386/t-crtstuff"
+    ;;
 avr-*-rtems*)
 	tmake_file="$tmake_file avr/t-avr avr/t-rtems t-fpbit"
 	tm_file="$tm_file avr/avr-lib.h"
diff -rupN gcc-4.8.0/libstdc++-v3/crossconfig.m4 gcc-4.8.0-patched/libstdc++-v3/crossconfig.m4
--- gcc-4.8.0/libstdc++-v3/crossconfig.m4	2013-05-27 00:06:18.413989202 +0200
+++ gcc-4.8.0-patched/libstdc++-v3/crossconfig.m4	2013-05-27 00:08:15.350544987 +0200
@@ -150,6 +150,12 @@ case "${host}" in
     GCC_CHECK_TLS
     AM_ICONV
     ;;
+  *-pirix*)
+   AC_CHECK_HEADERS([sys/types.h locale.h float.h])
+   GLIBCXX_CHECK_BUILTIN_MATH_SUPPORT
+   GLIBCXX_CHECK_COMPLEX_MATH_SUPPORT
+   GLIBCXX_CHECK_STDLIB_SUPPORT
+   ;;
   *-mingw32*)
     GLIBCXX_CHECK_LINKER_FEATURES
     GLIBCXX_CHECK_MATH_SUPPORT
